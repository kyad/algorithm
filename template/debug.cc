// vector
template<class T> ostream& operator<<(ostream& os, const vector<T> &v) {
  os << '[';
  for (size_t i = 0; i < v.size(); i++) {
    os << v[i] << (i == v.size() - 1 ? ']' : ' ');
  }
  return os;
}

// 2d-matrix
template<class T> ostream& operator<<(ostream& os, const vector<vector<T> > &v) {
  for (size_t i = 0; i < v.size(); i++) {
    os << (i == 0 ? "[[" : " [");
    for (size_t j = 0; j < v[i].size(); j++) {
      os << v[i][j] << (j == v[i].size() - 1 ? ']' : ' ');
    }
    os << (i == v.size() - 1 ? ']' : '\n');
  }
  return os;
}

// multiset
template<class T> ostream& operator<<(ostream& os, const multiset<T> &ms) {
  for (typename multiset<T>::iterator it = ms.begin(); it != ms.end(); it++) {
    os << *it;
    if (it != prev(ms.end())) {
      os << ' ';
    }
  }
  return os;
}

// deque
template<class T> ostream& operator<<(ostream& os, const deque<T> &dq) {
  for (auto it = dq.begin(); it != dq.end(); it++) {
    os << *it;
    if (it != prev(dq.end())) {
      os << ' ';
    }
  }
  return os;
}
