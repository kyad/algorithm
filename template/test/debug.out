(1, 2)
v=[1 2]
s=[1 2]
ms=[1 2 2]
mp=[{1: 2} {3: 10} {7: 20}]
q=[1 2]
pq=[5 2 1]
pqg=[1 2 5]
dq=[1 2]
st=[1 2]
(1, 2, 3)
76543210
00000101
mat=
[[1 2]
 [3 4]]
76543210  0 1 2 3 4
00000000 [1 1 1 1 1]
00000001 [1 1 1 1 1]
00000010 [1 1 1 1 1]
00000011 [1 1 1 1 1]
         0    1
   0 [   1    2]
   1 [   3    4]
v=[1 2]
s=[1 2]
v=[1 2]
s=[1 2]
ms=[1 2 2]
mp=[{1: 2} {3: 10} {7: 20}]
q=[1 2]
pq=[5 2 1]
dq=[1 2]
st=[1 2]
